src/Core/src/Configuration.php

```diff
@@ -40,16 +40,20 @@ class Configuration
     ];
 
```

Comment:
Maybe add a comment here to explain all these arrays?

> Put fallback options into groups to avoid mixing of provided config and environment variables